enhancements/cloud-integration/infrastructure-external-platform-type.md

```diff
@@ -0,0 +1,409 @@
+infrastructure related components it would change its semantic to signal when OpenShift is running on an infrastructure platform
+that will receive supplemental components from the infrastructure provider.
+
+### Implementation phases
+
+For make such change implementation process more smooth, it would be natural to split this activity to phases.
+The proposed phases breakdown can be found in [OCPPLAN-9429](https://issues.Red Hat.com/browse/OCPPLAN-9429).
+
+This document intended to describe initial phases of this project, proposed initial course of action:
+
+1. Update "openshift/api" with adding "External" PlatformType
+2. Ensure that all Red Hat operator tolerate the "External" platform and treat the same as "None" platform
+
+Next phase out of the scope for this EP:
+
+3. Modify operators for doing specific things when seeing "External" platform. Briefly it would be described down below, where possible for the moment.
+
+
+### Future work
+
+There are a number of topics in this area which would be wise to defer for upcoming phases, namely:
+
+1. Define missing capabilities and its concrete behaviour, for example add a "capability" for machine-api
+2. Precisely define reaction of the operators listed below for the "External" platform type 
+3. Define and document concrete mechanism for supplement cluster with provider-specific components during installation time (CCM, MAPI controller, DNS controller)
+
+### Affected components
+
+This section intended to enumerate OpenShift's components and briefly elaborate the future plans around this proposal.
+During initial implementation we must ensure that all OpenShift's components treat "External" platform in the same way as "None" for do not disrupt the current openshift operation.
+
+In the future we will need to change components behaviour to be able to receive supplemental provider-specific components from an infrastructure provider
+or, if a component manages something else (i.e. kubelet, kcm), adjust its behaviour (set `--cloud-provider=external` arg to kubelet for example).
+
+Specific component changes would be described in details within separate enhancement documents on a per-component basis.
+
+#### Machine Config Operator
+
+Currently, MCO sets kubelet flags based on a [PlatformStatus and PlatformType](https://github.com/openshift/machine-config-operator/blob/7babc03f84088f49072c6f5b0d1769850d09a694/pkg/controller/template/render.go#L343).
+Such flag is [crucial](https://kubernetes.io/docs/tasks/administer-cluster/running-cloud-controller/#running-cloud-controller-manager) for Cloud Controller Manager engagement withing the cluster.
+
+Initially, new "External" platform should be treated similarly to PlatformType "None" by the MCO, and do not set up any cloud-specific flags for the kubelet.
+Then, down the road, it would be expected for the MCO to use "External" platform type as a signal about underlying platform presence and operate accordingly.
```

Comment:
i think it would be nice to reference the "phase 3" part of timeline in addition to saying "down the road", so that people know how far down the road we are talking about. and perhaps mention that we might need changes to the PlatformSpec/Status.

Reply:
added notion in brackets with link to the respective section